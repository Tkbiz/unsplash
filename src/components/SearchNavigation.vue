<template>
  <v-container class="search-position pa-xl-0">
    <v-col cols="12"
           class="search-nav">
      <v-row>
        <v-text-field
            v-model="search"
            @change="searchPhotos"
            class="custom-input"
            hide-details
            label="Search for high-resolution photos">
          <v-img
              @click="searchPhotos"
              class="search-icon"
              slot="append"
              src="@/assets/images/icons/search.svg">
          </v-img>
        </v-text-field>
        <div class="trending">
          Trending
        </div>
        <v-chip-group
            active-class="primary--text"
            column>
          <v-chip
              v-for="(tag, index) in tags"
              :key="tag + index"
              @click="searchCollectionPhotos(tag)">
            {{ tag }}
          </v-chip>
        </v-chip-group>
      </v-row>
    </v-col>
  </v-container>
</template>

<script>
import {bus} from '@/bus'

export default {
  name: 'SearchNavigation',
  props: {
    tags: {
      type: Array,
      required: true
    }
  },
  data: () => ({
    search: null,
    page: 1
  }),
  methods: {
    async searchPhotos() {
      let tagName = this.search
      await this.$store.dispatch('searchPhotos', {
        tagName: tagName,
        page: this.page
      })
      bus.$emit('change-photos', this.photosNewList)
      this.$store.commit('fetchTagName', tagName)
      this.search = null
    },
    async searchCollectionPhotos(tag) {
      await this.$store.dispatch('searchPhotos', {
        tagName: tag,
        page: this.page
      })
      bus.$emit('change-photos', this.photosNewList)
      this.$store.commit('fetchTagName', tag)
    }
  },
  computed: {
    photosNewList() {
      return this.$store.getters.photos
    }
  }
}
</script>
