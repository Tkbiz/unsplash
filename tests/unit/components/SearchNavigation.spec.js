import {mount, createLocalVue} from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import SearchNavigation from '@/components/SearchNavigation'

Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(Vuex)

describe('SearchNavigation.vue', () => {
  let vuetify
  let wrapper
  let actions
  let mutations
  let getters
  let store

  beforeEach(() => {
    vuetify = new Vuetify()
    getters = {
      photosNewList: () => []
    }
    actions = {
      searchPhotos: jest.fn()
    }
    mutations = {
      fetchTagName: jest.fn()
    }
    store = new Vuex.Store({
      getters,
      actions,
      mutations
    })

    wrapper = mount(SearchNavigation, {
      localVue,
      vuetify,
      propsData: {
        tags: []
      },
      data: () => ({
        search: null,
        page: 1
      }),
      store
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders props.tags when passed', () => {
    const tags = []
    expect(wrapper.props()).toEqual({tags})
    expect(wrapper.props().tags).toEqual([])
  })

  it('when searchPhotos is called $emit is called', () => {
    wrapper.vm.$emit('change-photos', [])

    expect(wrapper.emitted('change-photos')).toBeTruthy()
    expect(wrapper.emitted('change-photos').length).toBe(1)
  })
})
