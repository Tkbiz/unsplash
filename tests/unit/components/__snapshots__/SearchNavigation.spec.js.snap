// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchNavigation.vue should match snapshot 1`] = `
<div class="container search-position pa-xl-0">
  <div class="search-nav col col-12">
    <div class="row">
      <div class="v-input custom-input v-input--hide-details theme--light v-text-field">
        <div class="v-input__control">
          <div class="v-input__slot">
            <div class="v-text-field__slot"><label for="input-15" class="v-label theme--light" style="left: 0px; position: absolute;">Search for high-resolution photos</label><input id="input-15" type="text"></div>
            <div class="v-input__append-inner">
              <div class="v-image v-responsive search-icon theme--light">
                <transition-stub name="fade-transition" mode="in-out">
                  <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
                </transition-stub>
                <div class="v-responsive__content"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="trending">
        Trending
      </div>
      <div class="v-item-group theme--light v-slide-group v-chip-group v-chip-group--column">
        <div class="v-slide-group__prev v-slide-group__prev--disabled">
          <transition-stub name="fade-transition"></transition-stub>
        </div>
        <div class="v-slide-group__wrapper">
          <div class="v-slide-group__content"></div>
        </div>
        <div class="v-slide-group__next v-slide-group__next--disabled">
          <transition-stub name="fade-transition"></transition-stub>
        </div>
      </div>
    </div>
  </div>
</div>
`;
